modules/alloy/src/main/ts/ephox/alloy/api/ui/Dropdown.ts

```diff
@@ -25,30 +26,98 @@ const factory: CompositeSketchFactory<DropdownDetail, DropdownSpec> = (detail, c
 
   const switchToMenu = (sandbox: AlloyComponent) => {
     Sandboxing.getState(sandbox).each((tmenu) => {
+      // This will highlight the menu AND the item
       TieredMenu.tieredMenu.highlightPrimary(tmenu);
     });
   };
 
   const action = (component: AlloyComponent): void => {
     const onOpenSync = switchToMenu;
-    DropdownUtils.togglePopup(detail, Fun.identity, component, externals, onOpenSync, DropdownUtils.HighlightOnOpen.HighlightFirst).get(Fun.noop);
+    DropdownUtils.togglePopup(
+      detail,
+      Fun.identity,
+      component,
+      externals,
+      onOpenSync,
+      HighlightOnOpen.HighlightMenuAndItem
+    ).get(Fun.noop);
   };
 
   const apis: DropdownApis = {
     expand: (comp) => {
       if (!Toggling.isOn(comp)) {
-        DropdownUtils.togglePopup(detail, Fun.identity, comp, externals, Fun.noop, DropdownUtils.HighlightOnOpen.HighlightNone).get(Fun.noop);
+        DropdownUtils.togglePopup(
+          detail,
+          Fun.identity,
+          comp,
+          externals,
+          Fun.noop,
+          HighlightOnOpen.HighlightNone
+        ).get(Fun.noop);
       }
     },
     open: (comp) => {
       if (!Toggling.isOn(comp)) {
-        DropdownUtils.togglePopup(detail, Fun.identity, comp, externals, Fun.noop, DropdownUtils.HighlightOnOpen.HighlightFirst).get(Fun.noop);
+        DropdownUtils.togglePopup(
+          detail,
+          Fun.identity,
+          comp,
+          externals,
+          Fun.noop,
+          HighlightOnOpen.HighlightMenuAndItem
+        ).get(Fun.noop);
       }
     },
+    refetch: (comp) => {
+      // Generally, the triggers for a refetch should make it so that the
+      // sandbox has been created, but it's not guaranteed, so we still handle the
+      // case where there isn't yet a sandbox.
+      const optSandbox = Coupling.getExistingCoupled(comp, 'sandbox');
+      return optSandbox.fold(
+        () => {
+          // If we don't have a sandbox, refetch is the same as open,
+          // except we return when it is completed.
+          return DropdownUtils.togglePopup(
+            detail,
+            Fun.identity,
+            comp,
+            externals,
+            Fun.noop,
+            HighlightOnOpen.HighlightMenuAndItem
+          ).map(Fun.noop);
```

Comment:
Should we extract the part up until `map`, `get`, `expand`, `close`, etc... here to avoid the duplication as there seems to be a fair bit going on here? For example:
```ts
const togglePopup = (comp: AlloyComponent, highlightMode: HighlightOnOpen) => DropdownUtils.togglePopup(
  detail,
  Fun.identity,
  comp,
  externals,
  Fun.noop,
  highlightMode
);
```
and then here it could just be:
```ts
return togglePopup(comp, HighlightOnOpen.HighlightMenuAndItem).map(Fun.noop);
```

Reply:
That sounds good

- [ ] refactor popup as suggested.

Reply:
Updated.